app/uk/gov/hmrc/thirdpartydeveloperfrontend/service/ApplicationActionService.scala

```diff
@@ -57,48 +57,46 @@ class ApplicationActionService @Inject() (
   def toApiSubscriptionStatusList(
       application: ApplicationWithSubscriptionData,
       subscriptionFieldDefinitions: Map[ApiContext, Map[ApiVersionNbr, Map[FieldName, SubscriptionFieldDefinition]]],
-      summaryApiDefinitions: Map[ApiContext, ApiData]
+      summaryApiDefinitions: List[ApiDefinition]
     ): List[APISubscriptionStatus] = {
 
-    def handleContext(context: ApiContext, cdata: ApiData): List[APISubscriptionStatus] = {
-      def handleVersion(version: ApiVersionNbr, vdata: ApiVersion): APISubscriptionStatus = {
+    def handleContext(apiDefinition: ApiDefinition): List[APISubscriptionStatus] = {
+      def handleVersion(apiVersionNbr: ApiVersionNbr, apiVersion: ApiVersion): APISubscriptionStatus = {
         def zipDefinitionsAndValues(): List[SubscriptionFieldValue] = {
-          val fieldNameToDefinition = subscriptionFieldDefinitions.getOrElse(context, Map.empty).getOrElse(version, Map.empty)
-          val fieldNameToValue      = application.subscriptionFieldValues.getOrElse(context, Map.empty).getOrElse(version, Map.empty)
+          val fieldNameToDefinition = subscriptionFieldDefinitions.getOrElse(apiDefinition.context, Map.empty).getOrElse(apiVersionNbr, Map.empty)
```

Comment:
Might be worth capturing context once as there are lots of apiDefintion.context refs in this block